### PR TITLE
chore: Update dev guide to include instructions on how to run tests and build sdk docs

### PR DESCRIPTION
**Pull request by alexlatchford**
As discussed in https://github.com/kubeflow/pipelines/pull/4218#issuecomment-667137505 just updating dev guide to include instructions for how to run common tasks locally and tips for contributing to SDK docs.

cc @Bobgy

### CODE REVIEW

File: developer_guide.md [hunk 1]
```diff
@@ -92,12 +92,61 @@ cd backend/src/ && go test ./...
 ### Frontend
 TODO: add instruction
 
-### DSL
+### Python SDK
 ```bash
-pip install ./dsl/ --upgrade && python ./dsl/tests/main.py
-pip install ./dsl-compiler/ --upgrade && python ./dsl-compiler/tests/main.py
+python -m venv /tmp/.env && source /tmp/.env/bin/activate # Optional, stops you installing dependencies globally.
```
Comment (Bobgy) [inline developer_guide.md:97]: I think we are suggesting miniconda in sdk install documentation, so we can just assume developer already have a virtual env, or make it an orthogonal section

Comment (alexlatchford) [reply developer_guide.md:97]: No problems, I can't seem to find that documentation would you mind linking me to it so I can add the link in the code? I'm not seeing any mentions of miniconda in this repo.

**Comment by alexlatchford** [comment]
> Thank you!
> We typically prefer dev documentation next to code in README.md. Would you mind moving them?
> And we can add pointers to this document for extra visibility

No problems, just to be clear you're wanting me to move the tests/SDK docs section to a new `./sdk/python/README.md` file? Then add a link from the main `README` to the new `README`?

**Comment by alexlatchford** [comment]
> > Thank you!
> > We typically prefer dev documentation next to code in README.md. Would you mind moving them?
> > And we can add pointers to this document for extra visibility
> 
> No problems, just to be clear you're wanting me to move the tests/SDK docs section to a new `./sdk/python/README.md` file? Then add a link from the main `README` to the new `README`?

Hey @Bobgy mind taking another look at my questions when you get a chance please?

**Comment by Bobgy** [comment]
@alexlatchford Sorry, I missed the notification.

> No problems, just to be clear you're wanting me to move the tests/SDK docs section to a new ./sdk/python/README.md file? Then add a link from the main README to the new README?

Exactly! and add a link from `developer_guide` to the new README

File: developer_guide.md (hunk 1 above)
Comment (Bobgy) [reply developer_guide.md:97]: It's mentioned in kubeflow/website doc repo: https://www.kubeflow.org/docs/pipelines/sdk/install-sdk/

**Comment by Ark-kun** [comment]
JFYI: VSCode supports automatic build and preview of the docs. You just open any .rst file (leading to a prompt to install an extension) and then you can open preview which automatically updates every time you save the .rst file.

**Comment by Bobgy** [comment]
Good to know, that will be so convenient

**Comment by Bobgy** [comment]
Sorry, my reply might have been a little late on this. @alexlatchford do you mind me recreating the PR?

**Pull request closed without merging**